### PR TITLE
Clarify how to return text-plain responses for answers others than Strings

### LINKED ISSUE #9669: Clarify how to return text-plain responses for answers others than Strings
see my conversation with @yawkat  https://github.com/micronaut-projects/micronaut-core/pull/9307#issuecomment-1645390538

Lets document how to return things such as BigDecimals, Boolean etc. as text/plain

### PR DESCRIPTION
**Pull request by wetted**
closes #9669

### CODE REVIEW

**Comment by wetted** [comment]
ping @sdelamo 

Is this what you want, i.e. along these lines with documentation illustrated by examples like this?

File: src/main/docs/guide/httpServer/jsonBinding.adoc [hunk 1]
```diff
@@ -153,3 +153,14 @@ jackson:
     useBigDecimalForFloats: true
 ----
 
+== Text Binding
+
+Micronaut Framework supports "text/plain" Content-Type as a way of binding to plain text. However, `text/plain` messages are more restrictive about allowed types. To return plain text responses for answers other than strings, manually call the object `toString()` method.
```
Comment (yawkat) [inline src/main/docs/guide/httpServer/jsonBinding.adoc]: more restrictive than what?

**Review by yawkat — APPROVED** [review]

File: src/main/docs/guide/httpServer/jsonBinding.adoc (hunk 1 above)
Comment (wetted) [reply src/main/docs/guide/httpServer/jsonBinding.adoc]: Maybe you can tell me. I copied what you wrote. 😉

https://github.com/micronaut-projects/micronaut-core/commit/37874c634202233f35b7c9376a5edfd5d49861f2#diff-2544d80c201711826bfd793aeda3dfec281d52b26c0bfa75ee20d019fc55f229:~:text=%3D%3D%3D%3D%20%60text/plain%60%20messages,g.%20number%20types).

Whatever you think is best to say is fine by me.

Comment (yawkat) [reply src/main/docs/guide/httpServer/jsonBinding.adoc]: oh compared to 3.x

Comment (wetted) [reply src/main/docs/guide/httpServer/jsonBinding.adoc]: I'll add that phrase for clarity. Thanks.

Comment (wetted) [reply src/main/docs/guide/httpServer/jsonBinding.adoc]: @yawkat I clarified the docs a little bit.

**Review by sdelamo — APPROVED** [review]
I have changed the copy slightly and simplified the embedded example in the docs.

**Pull request merged**